pybedlite/tests/test_overlap_detector.py

```diff
@@ -188,3 +193,61 @@ def test_construction_from_interval(bed_records: List[BedRecord]) -> None:
             assert new_record.strand is BedStrand.Positive
         else:
             assert new_record.strand is record.strand
+
+
+def test_arbitrary_interval_types() -> None:
```

Comment:
I will need a few more tests if reviews are favorable.

But this 1 test shows a (_hopefully unrealistic_) complex example which flexes most functionality.